src/index.js

```diff
@@ -29,70 +29,91 @@ function App() {
     // we only call this retrieving from localStorage once, because we only need to load from saved items when page first loads up, never again
 
     useEffect(() => {
-        localStorage.setItem(LOCAL_STORAGE_KEY_DATA, JSON.stringify(todos))
-    },[todos]); // this effect only fires if the todos array changes
+        localStorage.setItem(LOCAL_STORAGE_KEY_DATA, JSON.stringify(todos));
+    },[todos]); // this effect only fires if the todos array changes: which is toggle complete and remove tasks
 
-    function handleAddItem(e) {
+    const handleAddItem = () => {
         const currentInput =  toDoInput.current.value;
         const pickedDueDate = dueDateInput.current.value;
+        const currentTodos = [...todos];
 
-        if(currentInput && pickedDueDate) {
-            setToDos(prevList => {
-                return [...prevList, { id:uuidv4(), name: currentInput, due: pickedDueDate, completed: false}]
-            })
+        if(currentInput && pickedDueDate){ // will implement validations in next step
```

Comment:
Coding style. Space after `if`
```suggestion
        if (currentInput && pickedDueDate){ // will implement validations in next step
```